### PR TITLE
feat: encryption field

### PR DESCRIPTION
**Pull request by dream2023**
### This is a ...
- [x] New feature
- [ ] Improvement
- [ ] Bug fix
- [ ] Others

### Description 

image

image

image

### Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Add  encryption field|
| 🇨🇳 Chinese | 添加加解密字段|

### Docs

| Language   | Link |
| ---------- | --------- |
| 🇺🇸 English |      |
| 🇨🇳 Chinese |    |

### Checklists
- [ ] All changes have been self-tested and work as expected
- [ ] Test cases are updated/provided or not needed
- [ ] Doc is updated/provided or not needed
- [ ] Component demo is updated/provided or not needed
- [ ] Changelog is provided or not needed
- [ ] Request a code review if it is necessary

### CODE REVIEW

File: packages/plugins/@nocobase/plugin-field-encryption/src/server/plugin.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    this.db.registerFieldTypes({
+      encryption: EncryptionField,
+    });
+
+    this.db.registerOperators({
+      $encryptionEq,
+      $encryptionNe,
+    });
+
+    this.db.on('fields.beforeCreate', (model, field) => {
```
Comment (2013xile) [inline packages/plugins/@nocobase/plugin-field-encryption/src/server/plugin.ts:29]: `model.type`

File: packages/plugins/@nocobase/plugin-field-encryption/src/server/plugin.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      if (field.values?.type === 'encryption') {
```
Comment (2013xile) [inline packages/plugins/@nocobase/plugin-field-encryption/src/server/plugin.ts:30]: `model.type`

File: packages/plugins/@nocobase/plugin-field-encryption/src/server/encryption-field.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+      );
+    };
+  }
+
+  bind() {
+    super.bind();
+    // @ts-ignore
+    this.on('afterFind', this.findListener);
+    this.on('beforeCreate', this.writeListener);
+    this.on('beforeUpdate', this.writeListener);
```
Comment (2013xile) [inline packages/plugins/@nocobase/plugin-field-encryption/src/server/encryption-field.ts:70]: `beforeSave` = `beforeCreate` + `beforeUpdate`

**Review by 2013xile — CHANGES_REQUESTED** [review]
测试用例是不是还要补一些？搜索、解密、创建字段报错这些。

**Comment by dream2023** [comment]
> 测试用例是不是还要补一些？搜索、解密、创建字段报错这些。

肯定要写，还没文档、单测、e2e 今天写

**Pull request merged**